crates/sui-adapter/src/execution_engine.rs

```diff
@@ -226,18 +226,23 @@ fn execute_transaction<
         };
         execution_result
     });
-    if !gas_status.is_unmetered() {
-        temporary_store.charge_gas(gas_object_ref.0, &mut gas_status, &mut result, gas);
-    }
-    if !is_system {
-        #[cfg(debug_assertions)]
-        {
-            if !Mode::allow_arbitrary_values() {
-                // ensure that this transaction did not create or destroy SUI
-                temporary_store.check_sui_conserved().unwrap();
+    #[allow(clippy::collapsible_if)]
+    if protocol_config.gas_model_version() > 1 {
+        if !is_system {
```

Comment:
Double-checking that all system transactions are unmetered so this is equivalent to the old `if !gas_status.is_unmetered()`. Also, wondering if unifying this concepts would help @lxfind in his work to add conservation checks to change_epoch

Reply:
you are correct about being the same check. I figured organized this way should help anybody make progress on this